File: misc/math_puzzles/square.cpp
#include <iostream>
#include <stdio.h>
#include <cmath>
#include <algorithm>
#include <cstring>

using namespace std;

int main(){
	int t;
	cin>>t;
	while(t--){
		int l,b;
		cin>>l>>b;
		int prod = l*b;
		int maxi = 0;
		for(int i=1;i<=max(l,b);i++){
			int temp = i*i;
			if(prod % temp == 0)
				maxi = prod / temp;
		}
		cout<<maxi<<endl;
	}
	return 0;
}

File: misc/math_puzzles/square.py
import math
t = (int(raw_input()))
while t:
	l, b = [int(x) for x in raw_input().split()]
	prod = l*b
	maxi = 0
	#print res
	for i in range (1, l+1):
	#print "iam here"
		temp = i * i
		if prod % temp == 0:
			maxi = prod / temp	
	print "%d" %(maxi)
	t = t - 1
